File: 프로그래머스/3/43238. 입국심사/입국심사.java
class Solution {
    public long solution(int n, int[] times) {
        long max = times[0];
        for(int t : times){
            max = Math.max(t, max);
        }
        
        long left = 1;
        long right = max * n;
        
        while(left < right){
            long mid = (left + right) / 2;
            long cnt = 0;
            for(int t : times){
                cnt += mid/t;
            }
            if(cnt >= n){
                right = mid;
            }else{
                left = mid + 1;
            }
        }
        return right;
    }
}

File: 프로그래머스/3/43238. 입국심사/입국심사.js
function solution(n, times) {
    var max = 0;
    times.forEach(t => {
        max = Math.max(max, t);
    })
    
    var left = 1;
    var right = max * n;
    while(left < right){
        var mid = parseInt((left + right) / 2);
        
        let cnt = 0;
        times.forEach(t => {
            cnt += parseInt(mid / t);
        })
        
        if(cnt < n){
            left = mid + 1;
        }else{
            right = mid;
        }
    }
    
    return right;
}